proposals/lip-0045.md

```diff
@@ -1313,6 +1317,52 @@ def error(ccm: CCM, errorStatus: uint32) -> None:
     )
 ```
 
+#### isChannelActive
+
+The `isChannelActive` function checks that a chain (or the mainchain if the chain account does not exist) has status `CHAIN_STATUS_ACTIVE`. On the mainchain, it also checks the liveness condition for the chain.
+
+```python
+def isChannelActive(chainID: ChainID) -> bool:
+    # We do not consider the channel active if it is the own chain.
+    if chainID == ownChainAccount.chainID:
+        return False
+
+    # Processing on the mainchain.
+    if ownChainAccount.chainID == getMainchainID():
+        # Account has to exist on the mainchain.
+        if chainAccount(chainID) does not exist:
+            return False
+        
+        # Check if chain status is active.
+        if not chainAccount(chainID).status == CHAIN_STATUS_ACTIVE:
+            return False
+        
+        # Check liveness condition for active chains.
+        timestamp = timestamp of the block where this state transition is processed
+        if timestamp - chainAccount(chainID).lastCertificate.timestamp > LIVENESS_LIMIT:
+            return False
```

Comment:
Here I was thinking very concretely that this function should be called before attempting to send a CCM, to ensure that the sending will not fail because of the isLive check. Maybe we could improve the description of the function to clearly state this?

Reply:
I see that you even wrote in the description that it checks the liveness condition. Maybe you could add "This means that it will return `False` if the liveness condition is violated even when the status of the chain is (still) `CHAIN_STATUS_ACTIVE`" to make it even more clear. But it's not a must.

Reply:
@AndreasKendziorra Note that the same holds for `isLive()`. It checks liveness, but does not change the chain status in case liveness is violated.

Reply:
I was commenting on it because the function name suggests that it checks that a certain state entry has a certain value (`chainAccount(chainID).status == CHAIN_STATUS_ACTIVE`), and it may be very counterintuitive when it returns false even when this condition is fulfilled. But then I read that the description is mentioning it. So, it's fine.